src/realm/decimal128.cpp

```diff
@@ -0,0 +1,360 @@
+/*************************************************************************
+ *
+ * Copyright 2019 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ **************************************************************************/
+
+#include <external/IntelRDFPMathLib20U2/LIBRARY/src/bid_conf.h>
+#include <external/IntelRDFPMathLib20U2/LIBRARY/src/bid_functions.h>
+#include <realm/decimal128.hpp>
+#include <cstring>
+#include <stdexcept>
+#include <sstream>
+
+namespace {
+constexpr int DECIMAL_EXPONENT_BIAS_128 = 6176;
+constexpr int MAX_STRING_DIGITS = 19;
+} // namespace
+
+namespace realm {
+
+// This is a cut down version of bid128_from_string() from the IntelRDFPMathLib20U2 library.
+// If we can live with only 19 significant digits, we can avoid a lot of complex code
+// as the significant can be stored in w[0] only.
```

Comment:
Does this meet Mongo requirements?

Reply:
We can easily revert to the Intel version when this is required.